master/internal/configpolicy/task_config_policy_intg_test.go

```diff
@@ -16,14 +16,51 @@ import (
 	"github.com/determined-ai/determined/master/pkg/schemas/expconf"
 )
 
-func TestPriorityAllowed(t *testing.T) {
+func TestFindAllowedPriority(t *testing.T) {
+	require.NoError(t, etc.SetRootPath(db.RootFromDB))
+	pgDB, cleanup := db.MustResolveNewPostgresDatabase(t)
+	defer cleanup()
+	db.MustMigrateTestPostgres(t, pgDB, db.MigrationsFromDB)
+
+	// No priority limit to find.
+	_, found, err := findAllowedPriority(nil, model.ExperimentType)
+	require.NoError(t, err)
+	require.False(t, found)
+
+	// Priority limit set.
+	globalLimit := 10
+	user := db.RequireMockUser(t, pgDB)
+	addConstraints(t, user, nil, fmt.Sprintf(`{"priority_limit": %d}`, globalLimit), model.ExperimentType)
+	limit, found, err := findAllowedPriority(nil, model.ExperimentType)
+	require.Equal(t, globalLimit, limit)
+	require.True(t, found)
+	require.NoError(t, err)
+
+	// NTSC priority set.
+	configPriority := 15
+	invariantConfig := fmt.Sprintf(`{"resources": {"priority": %d}}`, configPriority)
+	addConfigs(t, user, nil, invariantConfig, model.NTSCType)
+	limit, _, err = findAllowedPriority(nil, model.NTSCType)
+	require.ErrorIs(t, err, errPriorityImmutable)
+	require.Equal(t, configPriority, limit)
+
+	// Experiment priority set.
+	configPriority = 7
+	invariantConfig = fmt.Sprintf(`{"resources": {"priority": %d}}`, configPriority)
+	addConfigs(t, user, nil, invariantConfig, model.ExperimentType)
+	limit, _, err = findAllowedPriority(nil, model.ExperimentType)
+	require.ErrorIs(t, err, errPriorityImmutable)
+	require.Equal(t, configPriority, limit)
+}
```

Comment:
Can we add some workspace tests too just so that we have full function coverage?

Reply:
Cool! We can leave this as is then